UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java

```diff
@@ -157,56 +178,51 @@ public String getRuleCombiningAlgorithmId() {
     }
 
     /**
-     * Retrieves the particular condition for the evaluation.
-     *
-     * In an UXACML policy we may have 3 types of conditions:
-     * pre, ongoing and post.
-     * In this function we basically want to extract only the condition
-     * in which we're interested in.
+     * In UXACML we may have 3 types of conditions: pre, ongoing and post.
+     * This function retrieves the policy with the required condition.
      *
      * @param conditionName
      *          the required condition
      * @return a copy of the policyType containing only the required condition
+     * @throws PolicyException
      */
     @Override
-    public PolicyWrapper getPolicy( String conditionName ) {
-        PolicyType clonedPolicyType = clonePolicyType();
-        /**
-         * This is the most delicate part of this function.
-         * If this list of objects contains a ruleType it must be analyzed
-         * in order to retrieve only the required condition.
-         * In any other case the object will be copied inside cloned list.
-         */
+    public PolicyWrapper getPolicyForCondition( String conditionName ) throws PolicyException {
+        PolicyType clonedPolicyType = almostClonePolicyType();
         List<Object> objectList = policyType.getCombinerParametersOrRuleCombinerParametersOrVariableDefinition();
         List<Object> clonedObjectList = clonedPolicyType.getCombinerParametersOrRuleCombinerParametersOrVariableDefinition();
+
         for( Object obj : objectList ) {
             RuleType ruleType = (RuleType) obj;
+            List<ConditionType> conditionList = ruleType.getCondition();
```

Comment:
here as well !!
Line 201 refers directly to ruleType.getCondition()

Reply:
yes ok, but I gave precedence to readability since this part is pretty messy